crafters/image/ImageReader/tests/test_imagereader.py

```diff
@@ -3,35 +3,95 @@
 
 import numpy as np
 from PIL import Image
+import pytest
 
 from .. import ImageReader
 
 
-def create_test_image(output_fn, size_width=50, size_height=50):
+def create_pil_image(size_width, size_height):
     from PIL import Image
-    image = Image.new('RGB', size=(size_width, size_height), color=(155, 0, 0))
-    with open(output_fn, "wb") as f:
+    image = Image.new('RGB', size=(size_width, size_height), color=(50, 100, 150))
+    return image
+
+
+def create_test_image(filename, size_width, size_height):
+    image = create_pil_image(size_width, size_height)
+    with open(filename, "wb") as f:
         image.save(f, 'jpeg')
 
 
 def test_io_uri():
     crafter = ImageReader()
-    tmp_fn = os.path.join(crafter.current_workspace, 'test.jpeg')
-    img_size = 50
-    create_test_image(tmp_fn, size_width=img_size, size_height=img_size)
-    test_doc = crafter.craft(buffer=None, uri=tmp_fn)
-    assert test_doc['blob'].shape == (img_size, img_size, 3)
+    filename = os.path.join(crafter.current_workspace, 'test.jpeg')
```

Comment:
It would be cool if you could use the tmpdir fixture, we are trying to slowly update our tests with better practices.

Reply:
nice idea, I changed it to use the tmpdir.